### DISCUSSION TITLE
how to custom spacing for width and height

### DISCUSSION THREAD

**Discussion by subframe7536** [unocss/unocss#2149] [Q&A] [answered]
[tailwind's config](https://www.tailwindcss.cn/docs/customizing-spacing) does not effect

[playground](https://uno.antfu.me/play/?html=FAHgJglgbgBAxgGwIYGcUF4BEAnA9rgF3gFo5YAjAc1KgHoBGABhgHdiCBTFIiAOwT4di5BLjgBrTAD42nbiFqQoU0EvjI0WPIRJkYVGg2ZsmzPgN5CRYyTOKmFSqUA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHFYALaDGb0YrAFxwA2kdpUIMWgBo6rVGNTMY0ALRH95mAF19Rly4A3YSgsYRBUfQAmAAZoqzjor1ovLysVOBh1VHDDZVVVZggxaANEdPy4ZgDDMyIAIwUgqAV-JtDwgEoO63KyNIq%2BGFQsZFze1VYwYWZsIjGKiqHWGBqY6LBeHoW%2B3tJ%2BndIOziA&options=N4IgzgLgTglgxhEAuaBXApgXyA)

**Comment by zyyv** [comment, marked as answer]
We needn't use `extend` key. If you use typescript, you could get a type warning.

```diff
- extend: {
    spacing: {
      test: '200px',
    },
- },
```

**Reply by subframe7536** [reply]
delete these two line still fail to effect🧐 [playground](https://uno.antfu.me/play/?html=MYVwTmCmB2AuBcAoAPAEwJYDcAEwA2AhgM5EC8ARGAPZWy4C0wOARgOaOYD0AjAAzYB3erEhE66aHgmR6zPFWABrcgD4hIsck4ZMKlDtyESFarQZNsbDj35C%2BvAA4APbBKnQZchcrX17WnT1kZjAgkL1IJwdIYAR9LENiMkoaOmAOS3YmG0F6ACZeRxc3aVl5JVV1UVgArCCDfCSTVPMWLK4%2BXPtnV0lSrwrff206xEQgA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHFYALaDGb0YrAFxwA2kdpUIMWgBo6rVGNTMY0ALRH95mAF19Rly4A3YSgsYRBUfQAmAAZoqzjor1ovLysVOBh1VHDDZVVVZggxaANEdPy4ZgDDMyIAIwUgqAV-JtDwgEoO63KyNIrWMGFmbCJc3tUYVFYYGpjosF4eitJ%2Bvs5SDs4gA&options=N4IgzgLgTglgxhEAuaBXApgXyA)

also, many features are undocumented, and it is too unfriendly for developers to explore the source code

**Reply by zyyv** [reply]
Because `width` rules doesn't use `spacing` of theme.

[Playground](https://uno.antfu.me/play/?html=FAHgJglgbgBAxgGwIYGcUF4BEAnA9rgF3gFo5YAjAc1KgHoBGABhgHdiCBTFIgW3a6IQAdgmEdi5BLjgBrTAD42nbiFqQo80OvjI0WPIRJkYVGg2ZsmzYaKHjJ0uYuJXV6%2BUA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHFYALaDGb0YrAFxwA2kdpUIMWgBo6rVGNTMY0ALRH95mAF19Rly4A3YSgsYRBUfQAmAAZoqzjor1ovLysVOBh1VHDDZVVVZggxaANEdPy4ZgDDMyIAIwUgqAV-JtDwgEoO63KyNIrWMGFmbCJc3tUYVFYYGpjosF4eitJ%2B-IB3YGRM8YrJ6dm6ecXl-NX089IOziA&options=N4IgzgLgTglgxhEAuaBXApgXyA)


https://github.com/unocss/unocss/blob/f662a521ec29e6b62095d80e4618d3e245072737/packages/preset-mini/src/_rules/size.ts#L16-L32
